avocado/core/runners/sysinfo.py

```diff
@@ -0,0 +1,311 @@
+        super().__init__(os.path.basename(path))
+        self.path = path
+
+    def __repr__(self):
+        r = "sysinfo.Logfile(%r, %r)"
+        r %= (self.path, self.logf)
+        return r
+
+    def __eq__(self, other):
+        if isinstance(other, _Logfile):
+            return (self.path, self.logf) == (other.path, other.logf)
+        elif isinstance(other, _Collectible):
+            return False
+        return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        return not result
+
+    def __hash__(self):
+        return hash((self.path, self.logf))
+
+    def run(self, queue, logdir):
+        """
+        Send the content of log file to the runner messages queue.
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param logdir: Path to a log directory.
+        :type logdir: str
+        """
+        if os.path.exists(self.path):
+            try:
+                with open(self.path) as f:
+                    lines = f.readlines()
+                logf_path = os.path.join(logdir, self.logf)
+                for line in lines:
+                    queue.put(messages.FileMessage.get(line, logf_path))
+            except IOError:
+                log.debug("Not logging %s (lack of permissions)", self.path)
+        else:
+            log.debug("Not logging %s (file does not exist)", self.path)
+
+
+class _Command(_Collectible):
+
+    """
+    Collectible command.
+
+    :param cmd: String with the command.
+    :type cmd: str
+    """
+
+    def __init__(self, cmd):
+        super().__init__(cmd)
+        self.cmd = cmd
+
+    def __repr__(self):
+        return "sysinfo.Command(%r, %r)" % (self.cmd, self.logf)
+
+    def __eq__(self, other):
+        if isinstance(other, _Command):
+            return (self.cmd, self.logf) == (other.cmd, other.logf)
+        elif isinstance(other, _Collectible):
+            return False
+        return NotImplemented
+
+    def __ne__(self, other):
+        result = self.__eq__(other)
+        if result is NotImplemented:
+            return result
+        return not result
+
+    def __hash__(self):
+        return hash((self.cmd, self.logf))
+
+    def run(self, queue, logdir):
+        """
+        Execute the command as a subprocess and send its output to
+        the runner messages queue.
+
+        :param queue: queue for the runner messages
+        :type queue: multiprocessing.SimpleQueue
+        :param logdir: Path to a log directory.
+        :type logdir: str
+        """
+        env = os.environ.copy()
+        config = settings.as_dict()
+        if "PATH" not in env:
+            env["PATH"] = "/usr/bin:/bin"
+        locale = config.get("sysinfo.collect.locale")
+        if locale:
+            env["LC_ALL"] = locale
+        timeout = config.get('sysinfo.collect.commands_timeout')
```

Comment:
We designed this to have always a default value, but sometimes we get  a None here.  So the next line you are doing an `int(None)` what will  raise  a `TypeError`.

Reply:
Right, I will fix that